src/cmd/build.rs

```diff
@@ -195,56 +217,130 @@ impl BuildCommand {
+                    ManifestPath::try_from(self.manifest_path.as_ref())?;
+                root_manifest_path
+                    .subcontract_manifest_path(package)
+                    .context(format!(
+                        "error: package ID specification `{}` did not match any packages",
+                        package
+                    ))?
+            }
+            None => ManifestPath::try_from(self.manifest_path.as_ref())?,
+        };
         let unstable_flags: UnstableFlags =
             TryFrom::<&UnstableOptions>::try_from(&self.unstable_options)?;
         let verbosity: Verbosity = TryFrom::<&VerbosityFlags>::try_from(&self.verbosity)?;
 
-        let args = ExecuteArgs {
-            manifest_path,
-            verbosity,
-            build_mode: BuildMode::Debug,
-            network: Network::default(),
-            build_artifact: BuildArtifacts::CheckOnly,
-            unstable_flags,
-            optimization_passes: OptimizationPasses::Zero,
-            keep_debug_symbols: false,
-            skip_linting: false,
-            output_type: OutputType::default(),
-        };
-
-        execute(args)
+        let mut check_results = Vec::new();
+
+        match self.check_all {
+            true => {
+                let workspace_members = get_cargo_workspace_members(&manifest_path)?;
+                for package_id in workspace_members {
+                    let manifest_path =
+                        util::extract_subcontract_manifest_path(package_id)
+                            .expect("Error extracting package manifest path");
+
+                    let args = ExecuteArgs {
+                        manifest_path,
+                        verbosity,
+                        build_mode: BuildMode::Debug,
+                        network: Network::default(),
+                        build_artifact: BuildArtifacts::CheckOnly,
+                        unstable_flags: unstable_flags.clone(),
+                        optimization_passes: OptimizationPasses::Zero,
+                        keep_debug_symbols: false,
+                        skip_linting: false,
+                        output_type: OutputType::default(),
+                    };
+                    check_results.push(execute(args)?);
+                }
+            }
+            false => {
+                let args = ExecuteArgs {
+                    manifest_path,
+                    verbosity,
+                    build_mode: BuildMode::Debug,
+                    network: Network::default(),
+                    build_artifact: BuildArtifacts::CheckOnly,
+                    unstable_flags,
+                    optimization_passes: OptimizationPasses::Zero,
+                    keep_debug_symbols: false,
+                    skip_linting: false,
+                    output_type: OutputType::default(),
+                };
+                check_results.push(execute(args)?);
+            }
+        }
+        Ok(check_results)
```

Comment:
Hmm so having all the results bundled together until the end results in some confusing output

```
❯ cargo contract check --all
 [1/2] Checking ink! linting rules
    Updating git repository `https://github.com/paritytech/ink/`
Checking with toolchain `nightly-2022-06-30-aarch64-apple-darwin`
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
 [2/2] Executing `cargo check`
    Updating crates.io index
    Finished release [optimized] target(s) in 0.20s
 [1/2] Checking ink! linting rules
    Updating git repository `https://github.com/paritytech/ink/`
Checking with toolchain `nightly-2022-06-30-aarch64-apple-darwin`
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
 [2/2] Executing `cargo check`
    Updating crates.io index
    Finished release [optimized] target(s) in 0.21s

Your contract's code was built successfully.

Your contract's code was built successfully.
```

We don't know which contract was built, and it just looks odd to have these two lines back to back. Let's try and add a bit more context here

Reply:
https://github.com/paritytech/cargo-contract/pull/647/commits/329ff8e23c5720a6efd00c41bdf4410423ea5a6d